core/src/main/java/com/netease/arctic/utils/ArcticDataFiles.java

```diff
@@ -12,22 +12,18 @@
 import java.math.BigDecimal;
 import java.net.URLDecoder;
 import java.nio.charset.StandardCharsets;
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.time.Instant;
-import java.time.OffsetDateTime;
+import java.time.LocalDateTime;
 import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
 import java.time.temporal.ChronoUnit;
 import java.util.Arrays;
-import java.util.Date;
 import java.util.List;
-import java.util.TimeZone;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
 public class ArcticDataFiles {
-  public static final OffsetDateTime EPOCH = Instant.ofEpochSecond(0).atOffset(ZoneOffset.UTC);
-  public static final SimpleDateFormat SDF = new SimpleDateFormat("yyyy-MM-dd-hh");
```

Comment:
The previous code used 'hh' to parse the hour, which may cause other bugs. 

I will test and evaluate the impact of this fix and determine whether it needs to be merged into versions 0.5.x and 0.6.x.

Reply:
> The previous code used 'hh' to parse the hour, which may cause other bugs.
> 
> I will test and evaluate the impact of this fix and determine whether it needs to be merged into versions 0.5.x and 0.6.x.

There should be no impact. There is no distinction between hh and HH in `SimpleDateFormat`, and both will be regarded as 24-hour format.

Reply:
I have built a corner case, when hour = 12.

```java
  public static void main(String[] args) throws ParseException {
    SimpleDateFormat sdf_hh = new SimpleDateFormat("yyyy-MM-dd-hh");
    sdf_hh.setTimeZone(TimeZone.getTimeZone("UTC"));
    Date date_hh = sdf_hh.parse("2022-08-08-12");
    System.out.println("date_hh = " + date_hh);

    SimpleDateFormat sdf_HH = new SimpleDateFormat("yyyy-MM-dd-HH");
    sdf_HH.setTimeZone(TimeZone.getTimeZone("UTC"));
    Date date_HH = sdf_HH.parse("2022-08-08-12");
    System.out.println("date_HH = " + date_HH);
  }
```
output
```
date_hh = Mon Aug 08 08:00:00 CST 2022
date_HH = Mon Aug 08 20:00:00 CST 2022
```

Reply:
I can be reproduced by changing the unit test case `com.netease.arctic.utils.TestArcticDataFiles#testHoursPartition`.

<img width="1194" alt="image" src="https://github.com/NetEase/amoro/assets/103108928/87307fb8-bd26-44d3-865d-af94ca578369">

Reply:
It seems true. yyyy-MM-dd-hh cannot parse the case where hour is 12. You need to specify AM or PM. If not specified, it will be parsed to midnight by default.
Then the previous logic is actually wrong.

Reply:
We should fix this bug immediately, is a backport required?

Reply:
Yes, I will help to cherry-pick to 0.5.x and 0.6.x after merging to master.